dev/com.ibm.ws.anno/src/com/ibm/wsspi/anno/info/MethodInfo.java

```diff
@@ -137,8 +137,18 @@ public interface MethodInfo extends Info {
      * @return The in-order parameter annotation collections of parameters
      *         of this method.
      */
+    @Deprecated
     public List<List<? extends AnnotationInfo>> getParameterAnnotations();
-
+    
+    /**
+     * <p>Answer the in-order parameter annotation collections of
+     * parameters of this method.</p>
+     *
+     * @return The in-order parameter annotation collections of parameters
+     *         of this method.
+     */
+    public List<? extends List<? extends AnnotationInfo>> getParmAnnotations();
```

Comment:
I have no idea why you are adding this method in and deprecating the one above it.  They are the same thing.  If you are going to add this method to the SPI class, then you need to update the minor version.  I just think this getParmAnnotations() stuff just should go away.

Reply:
List<? extends T> is different than List<T>.
The use of a wildcard provides a bound on the generic type without exactly specifying the type.  When used with a collection type, the effect is for the collection to be non-modifiable, because parameters are contravariant.
For this method, I think the new method was added because of how the annocache types extend the anno types.  The original getParameters could not be extended in the annocache copy of the type.